### ISSUE TITLE
Large multi-line regular expressions do not match

### ISSUE THREAD

**Issue by pillravi** [hawk/lux#22]
This seems to be a similar issue to #17, but on the expect side rather than the send side.

I am `cat`-ing a file multiple times to search for several different blocks of text. (I cannot use a file compare utility such as `diff` because the text blocks may appear in any order.) 

Script 3 passes:
```
!cat test.txt
"""??
<48 bytes (5 lines)>
"""
!cat test.txt
"""??
<48 bytes (5 lines)>
"""
!cat test.txt
"""??
<4057 bytes (113 lines)>
"""
!cat test.txt
"""??
<4057 bytes (113 lines)>
"""
```

while script 4 fails:
```
!cat test.txt
"""??
<Block 1: 48 bytes (5 lines)>
"""
!cat test.txt
"""??
<Block 2: 48 bytes (5 lines)>
"""
!cat test.txt
"""??
<Block 3: 3875 bytes (108 lines)>
"""
!cat test.txt
"""??
<Block 4: 3875 bytes (108 lines)>
"""
```
with the following output:
```
result            : FAIL at 40:21:-1:22:121:1834
expected
	<Block 3: 3875 bytes (108 lines)>
actual match_timeout

	<Block 3: 3875 bytes (108 lines)>
	<Block 4: 3875 bytes (108 lines)>
	SH-PROMPT:cat test.txt
	cat test.txt
	<Block 1: 48 bytes (5 lines)>
	<Block 2: 48 bytes (5 lines)>
	<Block 3: 3875 bytes (108 lines)>
	<Block 4: 3875 bytes (108 lines)>
	SH-PROMPT:cat test.txt
	<Block 1: 48 bytes (5 lines)>
	<Block 2: 48 bytes (5 lines)>
	<Block 3: 3875 bytes (108 lines)>
	<Block 4: 3875 bytes (108 lines)>
	SH-PROMPT:

successful        : 0
failed            : 1
script_04.lux:40:21:-1:22:121:1834 - match_timeout
summary           : FAIL
```

There is no particular length that causes it to fail; from the examples we see that an expect of 4057 bytes succeeds but an expect of 3875 bytes fails. However this problem does only manifest on expects that are qualitatively "long."

I am using the latest Lux from the `non-blocking-pty` branch.

**Comment by hawk** [comment]
Can you try tor reproduce it with the develop branch?

/Håkan

**Comment by pillravi** [comment]
Yes, the issue is reproducible on the develop branch. I see the new diff feature added with 1.16, but other than that the output is the same. Lux's diff seems to think there is something different in code block 3, but when I diff the `expected` and the `actual match_timeout` myself, they are identical.

**Comment by hawk** [comment]
Thanks.

How do I do to reproduce it? Do you have a executable test case?

/Håkan

**Comment by pillravi** [comment]
Here is an example test case:
[failing lux test.zip](https://github.com/hawk/lux/files/865928/failing.lux.test.zip)

**Comment by hawk** [comment]
Thanks for the test case. It reproduced the error.

It looks that there is an extra carriage return causing the problem. The
runpty program (in lux) reads and writes chunks of data. The extra CR
occurs when runpty reads a full buffer and its last char happens to be a
CR. Then the first char in the next read will be a spurious CR. I have not
figured out why that happens though.

/Håkan

**Comment by hawk** [comment]
It may work now. Can you re-run using lux on the develop-branch?

/Håkan
